### PR TITLE
Add back the coreclr fsharp unit test cases

### PR DESCRIPTION
**Pull request by KevinRansom**
This PR does a number of things related to the coreclr fsharp Cambridge test suite

1.  Use the coreclr for the test harness.
2.  Reactivate the invalidated test cases
3.  Execute tests using shared framework

ToDo:
Future PR's will extend the FSC test cases and enable the FSI Test cases.

### CODE REVIEW

File: tests/fsharp/FSharp.Tests.FSharpSuite.DrivingCoreCLR/Program.fs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module Program
```
Comment (brettfo) [inline tests/fsharp/FSharp.Tests.FSharpSuite.DrivingCoreCLR/Program.fs]: Copyright

**Comment by KevinRansom** [comment]
Appveyor Timed out.

File: tests/fsharp/FSharp.Tests.FSharpSuite.DrivingCoreCLR/Program.fs (hunk 1 above)
Comment (KevinRansom) [inline tests/fsharp/FSharp.Tests.FSharpSuite.DrivingCoreCLR/Program.fs]: Ok

Comment (KevinRansom) [inline tests/fsharp/FSharp.Tests.FSharpSuite.DrivingCoreCLR/Program.fs]: Hmmm we don't copyright the test files in the Cambridge suite. or at least not consistently.  I wonder why.

**Pull request merged**